packages/lib-user/src/components/shared/AllProjects/Projects/Projects.spec.js

```diff
@@ -0,0 +1,60 @@
+import { render, screen } from '@testing-library/react'
+import zooTheme from '@zooniverse/grommet-theme'
+import { Grommet } from 'grommet'
+
+import Projects from './Projects.js'
+import { PAGE_SIZE } from '../AllProjectsByCount/AllProjectsByCount.js'
+import { PROJECTS, PROJECT_CONTRIBUTIONS } from '../AllProjects.mocks.js'
+
+describe('components > shared > AllProjects', function () {
+  describe('when there are projectContributions', function () {
+
+    // Grommet's Pagination component is creating a `new Event()` out of our control.
+    // For this test suite, set global.Event to window.Event to prevent this error:
+    // Failed to execute 'dispatchEvent' on 'EventTarget': parameter 1 is not of type 'Event'.
+    // https://github.com/jsdom/jsdom/issues/3331#issuecomment-1137376839
+    before(function () {
+      global.Event = window.Event
+    })
+
+    after(function () {
+      // what to restore here?
+    })
```

Comment:
This works, and all lib-user tests pass, but I'm not sure what to restore `global.Event` to in `after()`. Do you know?

Reply:
It's probably the Node `Event` class:

https://nodejs.org/api/events.html#class-event

The safest way to handle globals is probably to store a reference to the old value, then restore that reference when you are done. 

```js
let oldEvent

// before your tests
oldEvent = global.Event

// after
global.Event = oldEvent
```

Personally, I recommend Playwright so that you run your browser code in a real browser, rather than trying to mock browser interfaces in Node with JSDOM.

Reply:
That's a good point to store the old event. I'll update the spec.

> It could be worth reading the conversation here:

Yes, that Issue is already include as a code comment in this spec file.